### DISCUSSION TITLE
Updating table values

### DISCUSSION THREAD

**Discussion by marcellintacite** [Akascape/CTkTable#81] [Q&A]
I am getting this error when updating values in the table :
![Screenshot 2024-03-25 122037](https://github.com/Akascape/CTkTable/assets/71480535/4379624c-5652-4c4c-8326-a0543b6f93aa)

Here's my code : 

`
def update_table(self):
        enseignant = EnseignantBackend(self.curseur)
        value = enseignant.allEnseignants()
        # Adding header
        header = ["No","Nom", "Matiere enseignee", "Numero de telephone","Utilisateur"]
        value.insert(0, header)        
        self.table.update_values(values=value)
`
Here's how i am calling this function initially

`
def afficherTableEnseignant(self):
        enseignant = EnseignantBackend(self.curseur)
        value = enseignant.allEnseignants()
        # Adding header
        header = ["No","Nom", "Matiere enseignee", "Numero de telephone","Utilisateur"]
        value.insert(0, header)
        self.table = CTkTable(self.enseignant_frame,row=10, column=5, values=value,command=self.show,header_color="#5999ff")
        self.table.grid(row=1, column=0, padx=5, pady=5, sticky="nsew")
`

**Comment by marcellintacite** [comment]
![r](https://github.com/Akascape/CTkTable/assets/71480535/dba86636-a66f-4163-9556-f084c0f0ee56)

**Comment by marcellintacite** [comment]
![Screenshot 2024-03-25 122501](https://github.com/Akascape/CTkTable/assets/71480535/58c9e657-d859-4927-998f-97a4fce15efb)

**Comment by Akascape** [comment]
@marcellintacite Give a minimal reproducible code for testing, I am not getting any error with update_values

**Reply by marcellintacite** [reply]
can i see how to update values, i am getting the error when using that function in my codebase

**Reply by marcellintacite** [reply]
Please i need your help 
Here's my codes, i am using OOP: 
![carbon](https://github.com/Akascape/CTkTable/assets/71480535/baf99a95-1fd6-4f2d-b3df-f749104d53f2)


I am getting this error : 

_**Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Users\hp\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1967, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "C:\Users\hp\AppData\Local\Programs\Python\Python312\Lib\site-packages\customtkinter\windows\widgets\ctk_button.py", line 554, in _clicked
    self._command()
  File "c:\Users\hp\Desktop\python\travail\ClassFrontend.py", line 142, in afficher_table_classe 
    self.table.update_values(values=self.table_values)
  File "C:\Users\hp\AppData\Local\Programs\Python\Python312\Lib\site-packages\CTkTable\ctktable.py", line 353, in update_values
    self.draw_table(**kwargs)
  File "C:\Users\hp\AppData\Local\Programs\Python\Python312\Lib\site-packages\CTkTable\ctktable.py", line 255, in draw_table**_

**Reply by Akascape** [reply]
@marcellintacite I am unable to reproduce this issue, make sure you are using the latest version. Or make a simple code which I can test and find the error. `update_values` method should work, I don't know why its not working in your case.
